### ISSUE TITLE
Where is Electron?

### ISSUE THREAD

**Issue by mckoss** [Microsoft/vscode#282]
When I run "code ." from the command line, I keep getting a dialog popping up saying "Where is Electron?" and asking me to locate it on disk.

- Why is it asking me this?
- What can I do about it?

![screen shot 2015-11-19 at 1 14 20 pm](https://cloud.githubusercontent.com/assets/82750/11284593/815e82ae-8ebf-11e5-8627-9e3e62bffaf5.png)

I'm running Version 0.10.1 on Mac OS.

**Comment by ainstushar** [comment]
How did u get code to run from terminal? I get "command not found." Did you create some sort of alias?

I created an alias "code" that points to the Electron executable inside the .app. ``alias code='/Applications/Visual\ Studio\ Code.app/Contents/MacOS/Electron'`` When I run ``code .`` it runs just fine.

**Comment by Tyriar** [comment]
Resolving this, our launch logic has changed significantly since this was created.

**Comment by ithinkihaveacat** [comment]
I've just encountered this on 1.1.1. Is there any workaround? The editor actually appears to work just fine, but the "Where is Electron?" dialog keeps reappearing every time it's dismissed.

**Comment by Tyriar** [comment]
@bpasero any ideas?

**Comment by bpasero** [comment]
No, never seen this.

**Comment by garysoed** [comment]
Running into this problem with 1.2.1 on Mac too.

**Comment by garysoed** [comment]
I can repro this reliably by clicking on the "Synchronize Changes" button. Here's an example log output:

```
git pull
git show HEAD:package.json
git show HEAD:package.json
156:164: execution error: Can’t get application "Electron". (-1728)
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
git show HEAD:package.json
git show HEAD:package.json
```

So a workaround to this would be to set `git.autofetch` to `false`.

**Comment by Tyriar** [comment]
May be to do with installation process? /cc @joaomoreno

**Comment by garysoed** [comment]
Maybe. I didn't do anything special about installing. Just downloaded the file, and opened the binary (I'm using Mac OS X El Capital).

**Comment by garysoed** [comment]
Moving Visual Studio Code out of and back into `Applications` didn't help.

**Comment by joaomoreno** [comment]
What does this running this in the Terminal return?

```
ls /Applications/Visual\ Studio\ Code.app/Contents/MacOS
```

**Comment by garysoed** [comment]
Just:
`Electron`

**Comment by christophberger** [comment]
Just if anyone still comes across this: I observed this behavior with version 1.4.0 on OSX El Capitan. I fixed it with this simple step:

When the "Where is Electron?" window popped up, I clicked the `Browse...` button and selected `/Applications/Visual Studio Code` (yes, VSC itself). After this, the annoying popups were gone.

I have no idea why this has worked, and maybe it only worked for me, so YMMV.
